core/block/template/service.go

```diff
@@ -1,5 +1,294 @@
 package template
 
-// It will be service soon...
+import (
+	"context"
+	"errors"
+	"fmt"
 
-const BlankTemplateID = "blank"
+	"github.com/anyproto/any-sync/app"
+	"github.com/anyproto/any-sync/commonspace/spacestorage"
+
+	"github.com/anyproto/anytype-heart/core/block"
+	"github.com/anyproto/anytype-heart/core/block/editor"
+	"github.com/anyproto/anytype-heart/core/block/editor/basic"
+	"github.com/anyproto/anytype-heart/core/block/editor/smartblock"
+	"github.com/anyproto/anytype-heart/core/block/editor/state"
+	"github.com/anyproto/anytype-heart/core/block/editor/template"
+	"github.com/anyproto/anytype-heart/core/block/export"
+	"github.com/anyproto/anytype-heart/core/block/object/idresolver"
+	"github.com/anyproto/anytype-heart/core/block/object/objectcreator"
+	"github.com/anyproto/anytype-heart/core/domain"
+	"github.com/anyproto/anytype-heart/pb"
+	"github.com/anyproto/anytype-heart/pkg/lib/bundle"
+	coresb "github.com/anyproto/anytype-heart/pkg/lib/core/smartblock"
+	"github.com/anyproto/anytype-heart/pkg/lib/database"
+	"github.com/anyproto/anytype-heart/pkg/lib/localstore/addr"
+	"github.com/anyproto/anytype-heart/pkg/lib/localstore/objectstore"
+	"github.com/anyproto/anytype-heart/pkg/lib/pb/model"
+	"github.com/anyproto/anytype-heart/space"
+	"github.com/anyproto/anytype-heart/util/internalflag"
+	"github.com/anyproto/anytype-heart/util/pbtypes"
+)
+
+const (
+	CName           = "template"
+	BlankTemplateID = "blank"
+)
+
+type Service interface {
+	StateFromTemplate(templateID, name string) (st *state.State, err error)
+	ObjectApplyTemplate(contextID string, templateID string) error
+	TemplateCreateFromObject(ctx context.Context, id string) (templateID string, err error)
+
+	TemplateCloneInSpace(space space.Space, id string) (templateID string, err error)
+	TemplateClone(spaceID string, id string) (templateID string, err error)
+
+	TemplateExportAll(ctx context.Context, path string) error
+
+	app.Component
+}
+
+type service struct {
+	picker       block.ObjectGetter
```

Comment:
it's better to use getblock package to reduce the number of transitive dependencies (block has huuuuge number of deps)

Reply:
Thank you for notice!
Done